### ISSUE TITLE
Default Error Produces Not in Schema

### ISSUE THREAD

**Issue by austins** [FastEndpoints/FastEndpoints#803]
I'm thinking this has to do with .NET 9, but I'll have to compare with .NET 8 if I get a chance.

Default error produces is supposed to be added by default when there is a validator for the request, but the resulting Swagger doc doesn't have ErrorResponse in the schema and the endpoint doesn't show as producing 400. This also repros if you switch to ProblemDetails.

If I breakpoint into here, the logic to add it does get run: https://github.com/FastEndpoints/FastEndpoints/blob/f786e67d49309519aeb0607f81ef169617286b15/Src/Library/Endpoint/Endpoint.Setup.cs#L671

I guess that it's possible that minimal API in .NET 9 could be removing it for some reason.

Here is a minimal repro:
- FE 5.30.0 and FE 5.30.0.15-beta
- .NET 9.0.100-rc.2
```cs
using FastEndpoints;
using FastEndpoints.Swagger;
using FluentValidation;

var builder = WebApplication.CreateBuilder(args);
builder
    .Services
    .AddFastEndpoints()
    .SwaggerDocument(
        o => o.DocumentSettings = s =>
        {
            s.Version = "v1";
            s.DocumentName = "v1";
        });

var app = builder.Build();
app.UseFastEndpoints();

if (app.Environment.IsDevelopment())
{
    app.UseSwaggerGen();
}

await app.RunAsync();

public sealed class MyEndpoint : Endpoint<MyRequest, MyResponse>
{
    public override void Configure()
    {
        Post("/myendpoint");
        AllowAnonymous();
    }

    public override Task HandleAsync(MyRequest req, CancellationToken ct)
    {
        return SendAsync(new MyResponse { Name = req.Name! }, cancellation: ct);
    }
}

public sealed class MyRequest
{
    public string? Name { get; init; }
}

public sealed class MyRequestValidator : Validator<MyRequest>
{
    public MyRequestValidator()
    {
        RuleFor(x => x.Name).NotEmpty();
    }
}

public sealed class MyResponse
{
    public required string Name { get; init; }
}
```

**Comment by dj-nitehawk** [comment]
copy/pasted the above code into a freshly scaffolded project via `dotnet new feproj -n Repro` and i see the 400 response documented in swagger.

![image](https://github.com/user-attachments/assets/3d9c44c1-8bcb-4e2a-bab3-04f258ffce69)

try this project: [Repro.zip](https://github.com/user-attachments/files/17529115/Repro.zip)

**Comment by austins** [comment]
Thanks. I see the 400 and FastEndpointsErrorResponse schema in your repro.

I just got a chance to compare the two versions. `<TargetFramework>net8.0</TargetFramework>` works fine, but `<TargetFramework>net9.0</TargetFramework>` is missing the 400 and FastEndpointsErrorResponse schema. Possible bug in .NET 9 or NSwag?

**Comment by dj-nitehawk** [comment]
did some digging and it's definitely a bug in nswag with .net 9.
even a minimal api endpoint like the following doesn't work:

```cs
app.MapPost("test", () => Results.Ok())
   .Produces<ErrorResponse>(400, "application/problem+json")
   .Produces<MyResponse>(200, "application/json");
```
nswag doesn't seem to be picking up anything other than the 200 produces metadata.
hopefully should get resolved when .net 9 comes out.

if you have the time, could you please see if it can be reproduced with a vanilla minimal api/nswag project on .net 9?

thanks!

**Comment by austins** [comment]
I can't repro it with a new vanilla minimal api/nswag project. 

[NSwagTest.zip](https://github.com/user-attachments/files/17529329/NSwagTest.zip)

![image](https://github.com/user-attachments/assets/1645ea06-f01b-4671-8d6c-00ce58e7fb49)

**Comment by austins** [comment]
This line in .NET 9's EndpointMetadataApiDescriptionProvider:
https://github.com/dotnet/aspnetcore/blob/8b857e42aee71f3d8030708e3a2d51191238f8fe/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L345C9-L345C130

Calls into this:
https://github.com/dotnet/aspnetcore/blob/8b857e42aee71f3d8030708e3a2d51191238f8fe/src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs#L233C13-L233C21

Which has this:
```cs
if (typeof(IResult).IsAssignableFrom(metadata.Type))
{
    continue;
}
```

When it gets to this line when the current metadata status code is 400 and type is FE's ErrorResponse or ProblemDetails, it continues and ends up being filtered out the resulting dictionary.

.NET 8 does not have this line of code. See: https://github.com/dotnet/aspnetcore/blob/49c3e206cdf565e780abae849304bd7110187bf5/src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs#L226

The [ErrorResponse](https://github.com/FastEndpoints/FastEndpoints/blob/ce139a6b84d8900a5a4e4962da2e5c53d8aade4d/Src/Library/DTOs/ErrorResponse.cs) and [ProblemDetails](https://github.com/FastEndpoints/FastEndpoints/blob/ce139a6b84d8900a5a4e4962da2e5c53d8aade4d/Src/Library/DTOs/ProblemDetails.cs) implement IResult, This could be what's happening.

**Comment by dj-nitehawk** [comment]
thanks for that info. was quite helpful. i've made a comment on an existing issue (possibly related) on the aspnet repo here: https://github.com/dotnet/aspnetcore/issues/57801#issuecomment-2439578287

let's see what they say...

**Comment by dj-nitehawk** [comment]
a workaround has been implemented in `v5.30.0.19-beta` 😉
